device/transport/mqtt/src/mqtt.ts

```diff
@@ -1002,7 +1003,9 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
     if (message.properties && message.properties.count() > 0) {
       for (let i = 0; i < message.properties.count(); i++) {
         if (i > 0 || sysPropString) topic += '&';
-        topic += encodeURIComponent(message.properties.propertyList[i].key) + '=' + encodeURIComponent(message.properties.propertyList[i].value);
+        const keyName = message.properties.propertyList[i].key;
+        topic += encodeURIComponent(((keyName === DT_SUBJECT) ? ('$.sub') : (keyName)))
```

Comment:
>'$.sub') [](http://example.com/codeflow?start=65&length=8)

It's strange that `$.sub` is treated differently than `$.cid`, etc., above.

Reply:
These things happen over time.  Lot of this code in here could probably be refactored to use more modern apis.